packages/gatsby-theme-carbon/src/components/ResourceCard/ResourceCard.js

```diff
@@ -1,112 +1,184 @@
+          )
+        : null}
+      {type === 'mini'
+        ? title && (
+            <h4 className={`${prefix}--resource-card__mini-title`}>{title}</h4>
+          )
+        : null}
+    </>
+  );
+
+  // This determines how and where the icons and image (children) are placed on the page.
+  const iconsAndImages = (
+    <>
+      {type === 'default'
+        ? icons &&
+          children && (
+            <>
+              <div className={`${prefix}--resource-card__icon--img`}>
+                {children}
+              </div>
+              <div className={`${prefix}--resource-card__icon--action`}>
+                {icons}
+              </div>
+            </>
+          )
+        : null}
+      {type === 'mini' ? (
+        children === undefined ? (
+          <div className={`${prefix}--resource-card__mini-icon--action`}>
+            {icons}
           </div>
-        </div>
-      </div>
+        ) : (
+          <div className={`${prefix}--resource-card__mini-icon--img`}>
+            {children}
+          </div>
+        )
+      ) : null}
+    </>
+  );
+
+  // This holds the subtitle, title, icons and images.
+  const cardContent = (
+    <>
+      {subTitle && (
+        <h5 className={`${prefix}--resource-card__subtitle`}>{subTitle}</h5>
+      )}
+      {titleSize}
+      {iconsAndImages}
+    </>
+  );
+
+  // This determines if the link that is supplied is staying internal to the site or if it's leaving the site.
+  let cardContainer;
+  if (disabled === true || href === undefined) {
+    cardContainer = <div className={carbonTileclassNames}>{cardContent}</div>;
+  } else if (isLink === true) {
+    cardContainer = (
+      <Link to={href} className={carbonTileclassNames} {...rest}>
+        {cardContent}
+      </Link>
+    );
+  } else {
+    cardContainer = (
+      <a href={href} className={carbonTileclassNames} {...rest}>
+        {cardContent}
+      </a>
     );
   }
-}
+
+  // This determines, based whether the card is default or mini, whether or not it gets the aspect-ratio-object class and associated div.
+  const aspectRatioObjectType = (
+    <>
+      {type === 'default' ? (
+        <div className={`${prefix}--aspect-ratio--object`}>{cardContainer}</div>
+      ) : null}
+      {type === 'mini' ? cardContainer : null}
+    </>
+  );
+
+  // This determines, based whether the card is default or mini, whether or not it gets the aspect-ratio and aspect-ratio-X:X classes and associated div.
+  const aspectRatioClassNames = (
+    <>
+      {type === 'default' ? (
+        <div
+          className={`${prefix}--aspect-ratio ${prefix}--aspect-ratio--${aspectRatio.replace(
+            ':',
+            'x'
+          )}`}
+        >
+          {aspectRatioObjectType}
+        </div>
+      ) : null}
+      {type === 'mini' ? aspectRatioObjectType : null}
+    </>
+  );
+
+  return (
+    <div {...rest} className={ResourceCardClassNames}>
+      {aspectRatioClassNames}
+    </div>
+  );
```

Comment:
Wondering if this too could be simplified. It can just be really confusing for maintainers when there are so many conditionals.

```suggestion
  // This determines, based whether the card is default or mini, whether or not it gets the aspect-ratio and aspect-ratio-X:X classes and associated div.
  const aspectRatioContainer = (
      {type === 'default' ? (
        <div
          className={`${prefix}--aspect-ratio ${prefix}--aspect-ratio--${aspectRatio.replace(
            ':',
            'x'
          )}`}
        >
          <div className={`${prefix}--aspect-ratio--object`}>
             {cardContainer}
           </div>
        </div>
      ) : cardContainer} // just gets cardContainer if type==="mini"
  );

  return (
    <div {...rest} className={ResourceCardClassNames}>
      {aspectRatioContainer}
    </div>
  );
```

Reply:
I would love to get on a call with you guys and brainstorm on this one.

Reply:
Yeah let's collab on this as a group, I think it's a great chance to practice together